lib/cdo/i18n_string_url_tracker.rb

```diff
@@ -0,0 +1,65 @@
+require 'concurrent'
+require 'cdo/firehose'
+
+class I18nStringUrlTracker
+  include Singleton
+  # the time (seconds) between each upload.
+  UPDATE_THREAD_PERIOD = 300 #5 minutes
+  # random variance for each loop.
+  UPDATE_THREAD_PERIOD_JITTER = 60 # 1 minute
+
+  def initialize
+    # store a buffer of data to send so updates can be batched together.
+    @buffer = Concurrent::Set[]
```

Comment:
what's the motivation for using `Concurrent::Set` specifically, rather than a basic `Set`? Is it just to make sure the buffer plays nicely with threads?

Reply:
years of techblog fearmongering about concurrency has driven this decision. I'm pretty sure that it's fine given that Ruby is supposed to be single threaded. But then, why did they add this data structure? 
If you feel this is a pre-optimization, I don't mind trying out a normal set first.

Reply:
Haha, fair enough. I think it makes sense to use this here, could you just add a comment explaining?

Reply:
On MRI (cruby) which we use, `Concurrent::Set` [is just an alias for `Set`](https://github.com/ruby-concurrency/concurrent-ruby/blob/082c05f136309fd7be56e7c1b07a4edcb93968f4/lib/concurrent-ruby/concurrent/set.rb#L25-L28). On other Ruby implementations (e.g., JRuby), there is no global VM lock across threads and so extra synchronization is needed to ensure thread safety.